### PR TITLE
 Merge shardingsphere-standalone-mode-repository-jdbc module(#25057)

### LINKED ISSUE #25057: Merge shardingsphere-standalone-mode-repository-jdbc module
Modify the module `shardingsphere-standalone-mode-repository-jdbc-core` by changing the current H2 and MySQL SPI to an XML configuration file that is directly load by the core module.

The XML attribute should contain all SQL statements of `JDBCRepositoryProvider` and the JDBC driver class name, which be parsed JAXB.

After the modification, `shardingsphere-standalone-mode-repository-jdbc-h2` and `shardingsphere-standalone-mode-repository-jdbc-mysql` can be deleted, and `shardingsphere-standalone-mode-repository-jdbc-core` should be renamed to "shardingsphere-standalone-mode-repository-jdbc" and moved up to the parent module.

The xml should be:

```xml
<sql type="H2" driver-class-name="xxx.xxx.xxx">
    <create-table> xxx</create-table>
    <select-by-key> xxx</select-by-key>
    <select-by-parent> xxx</select-by-parent>
    <insert> xxx</insert>
    <update> xxx</update>
    <delete> xxx</delete>
</sql>
```

### PR DESCRIPTION
**Pull request by iamhucong**
Fixes #25057.

Changes proposed in this pull request:
  - Use JAXB instead of SPI for loading JDBCRepositoryProvider.
  - Merge `shardingsphere-standalone-mode-repository-jdbc` module.

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [x] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: mode/type/standalone/repository/provider/jdbc/pom.xml [hunk 1]
```diff
@@ -25,11 +25,67 @@
...
+    </dependencies>
+    <build>
+        <testResources>
+            <testResource>
+                <directory>src/test/resources</directory>
+            </testResource>
+            <testResource>
+                <directory>src/main/resources</directory>
+            </testResource>
+        </testResources>
```
Comment (terrymanu) [inline mode/type/standalone/repository/provider/jdbc/pom.xml:89]: Is it necessary?

File: mode/type/standalone/repository/provider/jdbc/src/main/java/org/apache/shardingsphere/mode/repository/standalone/jdbc/provider/JDBCRepositoryProviderLoader.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Enumeration;
+import java.util.Objects;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+public class JDBCRepositoryProviderLoader {
```
Comment (terrymanu) [inline mode/type/standalone/repository/provider/jdbc/src/main/java/org/apache/shardingsphere/mode/repository/standalone/jdbc/provider/JDBCRepositoryProviderLoader.java:40]: Please add javadoc and `final` for the class

**Review by terrymanu — CHANGES_REQUESTED** [review]
Thank you for submit the codes, please fix the code review suggestions.

File: mode/type/standalone/repository/provider/jdbc/pom.xml (hunk 1 above)
Comment (iamhucong) [reply mode/type/standalone/repository/provider/jdbc/pom.xml:89]: This is to access and test H2.xml and MySQL.xml in unit testing.

Comment (terrymanu) [reply mode/type/standalone/repository/provider/jdbc/pom.xml:89]: If this configuration is removed, why can't src/main/resources/H2.xml and src/main/resources/MySQL.xml be accessed, and why can other modules access the XML without importing resources?

Comment (iamhucong) [reply mode/type/standalone/repository/provider/jdbc/pom.xml]: Hi, i have updated a submission, it turns out that this configuration is not necessary. The classpath already includes target/classes and target/test-classes in unit testing. Thank you for your guidance.

**Review by terrymanu — APPROVED** [review]

**Review by terrymanu — APPROVED** [review]

**Pull request merged**